sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITest.scala

```diff
@@ -902,6 +903,80 @@ class SparkE2EQueryITest
     item.getAs[String]("id") shouldEqual id
   }
 
+  "spark query" can "for single logical partition can be configured to use only one Spark partition" in  {
+    val cosmosEndpoint = TestConfigurations.HOST
+    val cosmosMasterKey = TestConfigurations.MASTER_KEY
+
+    val container = cosmosClient.getDatabase(cosmosDatabase).getContainer(cosmosContainer)
+
+    // assert that there is more than one range to ensure the test really is testing the parallelization of work
+    container.getFeedRanges.block().size() should be > 1
+    var lastId = ""
+    for (age <- 1 to 20) {
+      for (state <- Array(true, false)) {
+        val objectNode = Utils.getSimpleObjectMapper.createObjectNode()
+        objectNode.put("name", "Shrodigner's cat")
+        objectNode.put("type", "cat")
+        objectNode.put("age", age)
+        objectNode.put("isAlive", state)
+        lastId = UUID.randomUUID().toString
+        objectNode.put("id", lastId)
+        container.createItem(objectNode).block()
+      }
+    }
+
+    var cfg = Map("spark.cosmos.accountEndpoint" -> cosmosEndpoint,
+      "spark.cosmos.accountKey" -> cosmosMasterKey,
+      "spark.cosmos.database" -> cosmosDatabase,
+      "spark.cosmos.container" -> cosmosContainer,
+      "spark.cosmos.read.partitioning.strategy" -> "Restrictive"
+    )
+
+    // scalastyle:off underscore.import
+    // scalastyle:off import.grouping
+    import org.apache.spark.sql.types._
+    // scalastyle:on underscore.import
+    // scalastyle:on import.grouping
+
+    val customSchema = StructType(Array(
+      StructField("id", StringType),
+      StructField("name", StringType),
+      StructField("type", StringType),
+      StructField("age", IntegerType),
+      StructField("isAlive", BooleanType)
+    ))
+
+    var df = spark.read.schema(customSchema).format("cosmos.oltp").options(cfg).load()
+    var rowsArray = df.where(s"id = '$lastId'").collect()
+    rowsArray should have size 1
+    rowsArray(0).getAs[String]("id") shouldEqual lastId
+    df.rdd.getNumPartitions shouldEqual container.getFeedRanges.block().size()
+
+    spark.udf.register("GetFeedRangeForPartitionKey", new GetFeedRangeForPartitionKeyValue(), StringType)
+    val pkDefinition = "{\"paths\":[\"/id\"],\"kind\":\"Hash\"}"
```

Comment:
I wonder from the usability perspective, could we make it work if the user just passed `/id`? Knowing the full JSON contract might not be something users might be familiar with.

Reply:
Got it, as long as it's easy for the users to wire, makes sense